R-proj/src/InnerBall.cpp

```diff
@@ -0,0 +1,90 @@
+
+// Copyright (c) 20012-2018 Vissarion Fisikopoulos
+// Copyright (c) 2018 Apostolos Chalkis
+
+
+#include <Rcpp.h>
+#include <RcppEigen.h>
+#include <chrono>
+#include "cartesian_geom/cartesian_kernel.h"
+#include <boost/random.hpp>
+#include <boost/random/uniform_int.hpp>
+#include <boost/random/normal_distribution.hpp>
+#include <boost/random/uniform_real_distribution.hpp>
+#include "polytopes.h"
+#include "vpolyintersectvpoly.h"
+
+//' Compute an inscribed ball of a convex polytope
+//'
+//' For a H-polytope described by a \eqn{m\times d} matrix \eqn{A} and a \eqn{m}-dimensional vector \eqn{b}, s.t.: \eqn{Ax\leq b}, this function computes the largest inscribed ball (Chebychev ball) by solving the corresponding linear program.
+//' For a V-polytope \eqn{d+1} vertices that define a full dimensional simplex picked at random and the largest inscribed ball of the simplex is computed.
+//' For a zonotope \eqn{P} we compute the minimum \eqn{r} s.t.: \eqn{ r e_i \in P} for all \eqn{i=1, \dots ,d}. Then the ball centered at the origin with radius \eqn{r/ \sqrt{d}} is an inscribed ball.
+//'
+//' @param P A convex polytope. It is an object from class (a) HPolytope or (b) VPolytope or (c) Zonotope.
+//'
+//' @return A \eqn{d+1}-dimensional vector that describes the inscribed ball. The first \eqn{d} coordinates corresponds to the center of the ball and the last one to the radius.
+//'
+//' @examples
+//' # compute the Chebychev ball of a 2d unit simplex
+//' P = GenSimplex(2,'H')
+//' ball_vec = InnerBall(P)
+//'
+//' # compute the Chebychev ball of 3-dimensional cube in V-representation
+//' P = GenCube(3, 'V')
+//' ball_vec = InnerBall(P)
+//' @export
+// [[Rcpp::export]]
+Rcpp::NumericVector InnerBall(Rcpp::Reference P) {
+
+    typedef double NT;
+    typedef Cartesian<NT>    Kernel;
+    typedef typename Kernel::Point    Point;
+    typedef boost::mt19937    RNGType;
+    typedef HPolytope<Point> Hpolytope;
+    typedef VPolytope<Point, RNGType > Vpolytope;
+    typedef Zonotope<Point> zonotope;
+    typedef IntersectionOfVpoly<Vpolytope> InterVP;
+    typedef Eigen::Matrix<NT,Eigen::Dynamic,1> VT;
+    typedef Eigen::Matrix<NT,Eigen::Dynamic,Eigen::Dynamic> MT;
+    unsigned int n = P.field("dimension");
+
+    std::pair <Point, NT> InnerBall;
+
+    int type = P.field("type");
```

Comment:
why not using something like `std::is_same<T,U>::value` to test if type of `P` is `Hpolytope` etc and get rid of `type` variable?